pom.xml

```diff
@@ -230,6 +230,19 @@ THE SOFTWARE.
         <artifactId>access-modifier-annotation</artifactId>
         <version>${access-modifier-annotation.version}</version>
       </dependency>
+
+      <!--  make sure these plugins are never used by us or by any plugins which end up depening on this version -->
+
+      <dependency>
+        <!-- Jenkins test Harness tries to fudge this, causes issues with some IDEs -->
+        <groupId>javax.servlet</groupId>
+        <!-- the old artifactID for the servlet API -->
+        <artifactId>servlet-api</artifactId>
+        <version>[0]</version>
```

Comment:
What does this mean, if not `0`? A comment would be in order.

Reply:
This is a hard version range dependency that can only be satisfied by one and only one version, `0`

When you do `<version>0</version>` you are telling Maven
> Use any version you like, but I think `0` would be a good idea

The "closest" suggestion wins.

As soon as you have some version ranges specified, then Maven starts collating the intersection set... so if we had `<version>[0,1.2)</version>` and a dependency had `<version>[1,1.3]</version>` then Maven will conclude that the selected version should be in the range `[1,1.2)`... now if the closest suggestion was `1.0.1` and that suggestion is valid for the required range... Maven *should* use that suggestion, but otherwise it just picks from the range (in a way that may be documented but I cannot recall)

Keep in mind, that `<optional>` dependencies are allowed to contribute to range requirements... they just do not force their way onto the classpath.

By specifying `[0]` we force this version over anything else. 
* If somebody has specified a version in the normal way... requirements always beat suggestions and `0` wins.
* If somebody has specified a version range... if there is no intersection their build will fail... if there is an intersection including `0` then that wins.

Reply:
OK…anyway, my comment stands: a comment in the source code explaining this would be appropriate.

Reply:
@jglic comment left